app/elm/Routing.elm

```diff
@@ -37,3 +37,8 @@ imagePath id =
 personPath : PersonId -> String
 personPath id =
     "#/person/" ++ (toString id)
+
+
+personImagePath : ImageId -> PersonId -> String
+personImagePath imageId personId =
+    "#/person/" ++ (toString personId) ++ "/image/" ++ (toString imageId)
```

Comment:
Another approach you can take for this sort of thing rather than having separate path functions is to have a single function called something like `pathFor` that takes a `Route` as an argument:

```elm
pathFor route =
    case route of
        PersonRoute id ->
             "#/person/" ++ (toString id)
        PersonImageRoute personId imageId ->
             "#/person/" ++ (toString personId) ++ "/image/" ++ (toString imageId)
      ...
```

```elm
a [ href (routeFor (PersonRoute person.id)) ] [ text person.name ]
```

That's kind of nice in that if you ever change the arguments a route takes, the compiler will raise errors in all the places where you're using the route to generate the path.